projects/VitalRecord/VitalRecord_util.cs

```diff
@@ -94,30 +94,19 @@ protected void UpdateEntry(bool shouldExist, [CallerMemberName] string propertyN
             }
         }
 
-        // TODO: How can we make this flexible to support more types?
+
         /// <summary>Helper to support vital record property getter helper methods for values stored in Observations.</summary>
         /// <param name="code">the code to identify the type of Observation</param>
         /// <param name="extensionURL">if present, specifies that the value should be get from an extension with the provided URL instead</param>
-        protected Dictionary<string, string> GetObservationValue(string code, string extensionURL = null)
+        protected Observation GetObservation(string code)
```

Comment:
Seems like it could simplify a lot of the setters if this was GetOrCreateObservation?

Reply:
changed name and updated setters